### PR TITLE
frontend/fix: #1792 driver_crashlytics_fixes

### LINKED ISSUE #1792: Driver Crashlytics fixes
**Describe the bug**
Driver Crashlytics fixes

**To Reproduce**
Steps to reproduce the behavior.

**Expected behavior**

**Screenshots**

**App (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by PraveenGongada**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Fixed some crashlytics crashes for driver app

Closes #1792

### CODE REVIEW

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java [hunk 1]
```diff
@@ -2254,6 +2254,8 @@ public void storeCallBackLocateOnMap(String callback) {
     @JavascriptInterface
     public void showMap(final String pureScriptId, boolean isEnableCurrentLocation, final String mapType, final float zoom, final String callback) {
         try {
+            View view = activity.findViewById(Integer.parseInt(pureScriptId));
```
Comment (Rohit4997) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java:2257]: As this view we are not using These two lines can we replaced with the following
`if ((activity.findViewById(Integer.parseInt(pureScriptId)) == null) return;`

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java [hunk 2]
```diff
@@ -2272,6 +2274,8 @@ public void showMap(final String pureScriptId, boolean isEnableCurrentLocation,
     @JavascriptInterface
     public void mapSnapShot(final String pureScriptId, final String json, final String routeType, final boolean actualRoute, final String callback) {
         try {
+            View view = activity.findViewById(Integer.parseInt(pureScriptId));
```
Comment (Rohit4997) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java:2277]: As this view we are not using These two lines can we replaced with the following
`if ((activity.findViewById(Integer.parseInt(pureScriptId)) == null) return;`

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java [hunk 3]
```diff
@@ -1337,13 +1337,15 @@ private void encodeImageToBase64(@Nullable Intent data) {
...
-            handler.postDelayed(() -> {
+            handler.post(() -> {
                 try {
-                    inAppNotification.generateNotification(title, message, onTapAction, action1Text, action2Text, action1Image, action2Image, channelId, durationInMilliSeconds);
-                } catch (JSONException e) {
+                    if(inAppNotification != null) {
+                        inAppNotification.generateNotification(title, message, onTapAction, action1Text, action2Text, action1Image, action2Image, channelId, durationInMilliSeconds);
+                    }
+                } catch (Exception e) {
                     Log.e(TAG, "Error in In App Notification Handler " + e);
```
Comment (Rohit4997) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java:1346]: Can you log firebase event for the crash

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/InAppNotification.java [hunk 4]
```diff
@@ -168,10 +168,14 @@ private void attachEventListenerToNotification(String onTapAction) {
...
-                        String javascript = String.format(Locale.ENGLISH, "window.callUICallback('%s');", onTapAction);
-                        dynamicUII.addJsToWebView(javascript);
+                    try {
+                        DuiCallback dynamicUII = MainActivity.getInstance().getJuspayServices().getDuiCallback();
+                        if (dynamicUII != null && onTapAction != null) {
+                            String javascript = String.format(Locale.ENGLISH, "window.callUICallback('%s');", onTapAction);
+                            dynamicUII.addJsToWebView(javascript);
+                        }
+                    } catch (Exception e) {
+                        Log.e(LOG_TAG, "Error in onClick " + e);
```
Comment (Rohit4997) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/InAppNotification.java:178]: Can you log firebase event for the crash

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/LocationUpdateService.java [hunk 5]
```diff
@@ -981,16 +981,20 @@ public void onFailure(@NonNull Exception e) {
...
+                timer.cancel();
+                timer.purge();
+            }
+            if (timerTask != null)
+                timerTask.cancel();
+            timer = null;
+            timerTask = null;
+            isLocationUpdating = false;
+        } catch (Exception e) {
+            Log.e(LOG_TAG, "Error in cancelTimer " + e);
```
Comment (Rohit4997) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/LocationUpdateService.java:996]: Add a firebase log event with tag `EXCEPTION`

**Review by Rohit4997 — CHANGES_REQUESTED** [review]

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]
Add firebase logs for all the catch block

**Review by Vignesh-772 — APPROVED** [review]

**Review by Rohit4997 — APPROVED** [review]

**Pull request merged**